src/main/java/org/jenkinsci/remoting/protocol/IOHub.java

```diff
@@ -423,7 +423,13 @@ public final void unregister(SelectableChannel channel) {
     }
 
     private String getThreadNameBase(String executorThreadName) {
-        return "IOHub#" + _id + ": Selector[keys:" + selector.keys().size() + ", gen:" + gen + "] / " + executorThreadName;
+        int keySize;
+        try {
+            keySize = selector.keys().size();
+        } catch (ClosedSelectorException x) {
```

Comment:
Unrelated bugfix IIUC

Reply:
See the PR description. There were two exceptions. I fixed both.